### DISCUSSION TITLE
Is there a plan to add the generation of Oneof in generateproto?

### DISCUSSION THREAD

**Discussion by xaviave** [socotecio/django-socio-grpc#277] [General]
Currently using the [Oneof](https://protobuf.dev/programming-guides/proto3/#oneof) field in my proto file and i would like to use it in django and this package.
According to my search, it could be the [django-polymorphic](https://pypi.org/project/django-polymorphic/) package that help to do this kind of inheritance with models.
Do you have any plan to integrate it?

Really nice handling of the grpc with django seamlessly, i really like this package :)

**Comment by legau** [comment]
Hi @xaviave,
Integrating OneOf (and completing the proto3 specs) is in our roadmap #219 
If you have a specific use case that we can work against please add a comment in this issue to further help us designing it
As a first step towards the implementation I think we would allow it as a declared type parameter in the @grpc_action decorators
